### PR TITLE
[MRG] Added plot sets functionality

### PR DESCRIPTION
**Pull request by kmilo9999**
- Added plot sets to the Layout templates
- Added logic to 'Make Figure' button to handle data depended layouts 
solves #711 
drive-dipole
drive-spikes
dipole-layers
dipole-spectrogram
drive-dipole-spectrogram
psd-layers
image

### CODE REVIEW

**Comment by kmilo9999** [comment]
@gtdang give it a quick review and let me know your thoughts

File: hnn_core/gui/_viz_manager.py
Comment (gtdang) [inline hnn_core/gui/_viz_manager.py:781]: Since Dipole-Drive is the new default plot at the start I think the visibility should be "visible" and only hidden if a blank template is selected.

File: hnn_core/gui/_viz_manager.py
Comment (gtdang) [inline hnn_core/gui/_viz_manager.py:889]: I noticed that if `is_data_template` = True and `num_data_sims` == 0, a blank plot still is added and you can't get rid of it because the figure controls don't populate. 

Perhaps we should wrap `_add_figure` in a conditional so nothing is generated if certain conditions are not met?

File: hnn_core/gui/gui.py
Comment (gtdang) [inline hnn_core/gui/gui.py:1146]: Let's set this default to `"Drive-Dipole"`, the first template in the list.

**Review by gtdang — CHANGES_REQUESTED** [review]
Looking good Camilo. Left some suggestions and questions.

File: hnn_core/gui/gui.py
Comment (gtdang) [reply hnn_core/gui/gui.py:1146]: We discussed and we are keeping this as-is.

File: hnn_core/gui/_viz_manager.py
Comment (kmilo9999) [reply hnn_core/gui/_viz_manager.py]: I refactored the add_figure code. It validates the existence of data to create figures.

File: hnn_core/gui/_viz_manager.py
Comment (kmilo9999) [reply hnn_core/gui/_viz_manager.py]: I set the default visibility as `visible` as a default

File: hnn_core/gui/_viz_manager.py
Comment (kmilo9999) [reply hnn_core/gui/_viz_manager.py]: I applied your suggestion

File: hnn_core/gui/_viz_manager.py
Comment (kmilo9999) [reply hnn_core/gui/_viz_manager.py]: I refactored the `kwargs` property in the layout templates dictionaries. let me know your thoughts.

File: hnn_core/gui/_viz_manager.py
Comment (jasmainak) [inline hnn_core/gui/_viz_manager.py:871]: public function is missing docstring

File: hnn_core/gui/_viz_manager.py
Comment (jasmainak) [inline hnn_core/gui/_viz_manager.py:84]:
```suggestion
    "Drive-Dipole (2 x 1)": {
```

I kind of like knowing how many subplots I'll get ... but it's a personal preference

File: hnn_core/gui/_viz_manager.py
Comment (jasmainak) [inline hnn_core/gui/_viz_manager.py:121]:
```suggestion
    "PSD Layers (3 x 1)": {
```

**Comment by jasmainak** [comment]
@gtdang I let you iterate with @kmilo9999 and merge when happy!

Don't forget to update `whats_new` and would recommend keeping the commit history clean (< 10 commits)

File: hnn_core/gui/_viz_manager.py
Comment (jasmainak) [inline hnn_core/gui/_viz_manager.py:147]: public function with no docstring

File: hnn_core/gui/_viz_manager.py
Comment (jasmainak) [inline hnn_core/gui/_viz_manager.py:919]: no tests to be updated?

File: hnn_core/gui/_viz_manager.py
Comment (kmilo9999) [reply hnn_core/gui/_viz_manager.py]: changed to private

File: hnn_core/gui/_viz_manager.py
Comment (kmilo9999) [reply hnn_core/gui/_viz_manager.py]: I added the subplot grid spec to the layout template name

File: hnn_core/gui/_viz_manager.py
Comment (kmilo9999) [reply hnn_core/gui/_viz_manager.py]: I added the subplot grid spec to the layout template name

File: hnn_core/gui/_viz_manager.py
Comment (kmilo9999) [reply hnn_core/gui/_viz_manager.py]: Changed to private

**Comment by kmilo9999** [comment]
@gtdang @ntolley Please review the test I included in `test_gui.py` about checking the creation of plots with using the new templates.

File: hnn_core/tests/test_gui.py
Comment (gtdang) [inline hnn_core/tests/test_gui.py:341]: Nice test!

File: hnn_core/gui/_viz_manager.py
Comment (gtdang) [inline hnn_core/gui/_viz_manager.py:721]: Yeah I like the better! Should avoid evals where possible.

File: hnn_core/gui/_viz_manager.py
Comment (gtdang) [inline hnn_core/gui/_viz_manager.py:920]:
```suggestion
        # Plot data if it is a data-dependent template
```

**Review by gtdang — CHANGES_REQUESTED** [review]
Thanks Camilo. This is looking really great! Make sure to update the whats_new.rst file mentioning this new feature.

**Review by gtdang — CHANGES_REQUESTED** [review]
Suggested slight change to the what's new text to be more specific.

File: doc/whats_new.rst
Comment (gtdang) [inline doc/whats_new.rst:53]:
```suggestion
  by `Camilo Diaz`_ in :gh:`746`.
```

**Review by gtdang — APPROVED** [review]
Looks good to me!

**Comment by kmilo9999** [comment]
Awesome. Let me clean the commit history before merging with main

**Comment by kmilo9999** [comment]
@gtdang This one is ready to go! 👍

**Comment by kmilo9999** [comment]
@gtdang  I added the Dipole-Spike plot. Review it if you want. It is ready to go.

**Comment by gtdang** [comment]
@ntolley 
This is ready to merge if you're OK with it.

**Comment by ntolley** [comment]
@kmilo9999 it seems like there are rebase conflicts? Happy to merge once those are fixed (i.e. running `git rebase master` on the most recent version of the master branch, and fixing the merge conflicts).

We can discuss in our Thursday meeting if there's too many conflicts

**Comment by kmilo9999** [comment]
> @kmilo9999 it seems like there are rebase conflicts? Happy to merge once those are fixed (i.e. running `git rebase master` on the most recent version of the master branch, and fixing the merge conflicts).
> 
> We can discuss in our Thursday meeting if there's too many conflicts

It was a small conflict in the `whats_new.rst` file. I resolved and it should be ready to merge after checks.

**Comment by jasmainak** [comment]
@kmilo9999 please fetch master branch again and rebase ... it will get rid of the weird commits

File: doc/whats_new.rst
Comment (jasmainak) [inline doc/whats_new.rst:50]: This shouldn't be in your PR ... it's already down below

**Comment by kmilo9999** [comment]
@ntolley It should be ready to merge.

File: hnn_core/gui/_viz_manager.py
Comment (ntolley) [reply hnn_core/gui/_viz_manager.py:154]: @kmilo9999 do you think this is sufficiently covered by the current unit tests? It's a pretty trivial function so I'll leave it to your judgement

Comment (kmilo9999) [reply hnn_core/gui/_viz_manager.py:154]: > @kmilo9999 do you think this is sufficiently covered by the current unit tests? It's a pretty trivial function so I'll leave it to your judgement

Yes, the unit test already covers this function inside the `gui.viz_manager.make_fig_button`  call.

**Review by ntolley — APPROVED** [review]
Great work @kmilo9999!! This is going to be a superl feature for the GUI

**Pull request merged**
